### ISSUE TITLE
ImGui w/ osgearth

### ISSUE THREAD

**Issue by quentinBdr** [gwaldron/osgearth#2069]
Hi,

I'm running osgearth v3.3 and osg v3.6.5 on win10. 
In my application which is based on Qt, the imgui ApplicationGUI is displayed as follow :

![image](https://user-images.githubusercontent.com/15386075/169493218-f152ffb3-1c1d-4643-a77c-a65ab9309427.png)

My problem is, the mouse is not taken into account by imgui. Although I can move around in osgearth normally, the mouse interactions does not affect imgui. I can see that mouse events are passing through the OsgImGuiHandler::handle, but the attribute io.WantCaptureMouse stays at false.

I've implemented imgui just like in the osgearth_imgui application provided. I've tried running the osgearth_imgui app but no imgui shows up.

What am I missing in my application that sort of disable the mouse interaction on imgui ?

**Comment by quentinBdr** [comment]
Here is what I've got after disabling fullscreen (got 3 screens) :
![image](https://user-images.githubusercontent.com/15386075/169760223-d5cd86dc-4323-425a-8456-8aa4169da840.png)

As for my issue, I was able to track deeper where the problem was. It turns out the mouse coordinates that were given to imgui got a negative Y (e.g. (1200, -920)). So ImGui did not detect a hovering on its context window, hence never set to true the io.WantCaptureMouse bool.
That leads me to another question, why the Y is negative, when in the Qt part I do pass in a positive Y ? By passing -Y from Qt, ImGui is working well, but the EarthManipulator got its axis mixed up. I struggle to understand why the Y is being inverted somewhere.

**Comment by gwaldron** [comment]
I'm afraid I have no idea 🤷‍♂️ but if you can figure it out, feel free to submit a PR with a fix for all to use! Thanks.

**Comment by chrisjaquet** [comment]
I do not know if this is related, but when integrating an alternative to ImGUI into our app we needed to explicitly check the direction of the Y-Axis before trying to manipulate things in viewport coordinates. Something like:
```
// Our viewport coordinates are X increasing left and Y increasing downwards, which is the opposite of OSG
bool isYAxisInverted = (pViewerView->getEventQueue()->getCurrentEventState()->getMouseYOrientation()
    == osgGA::GUIEventAdapter::Y_INCREASING_DOWNWARDS);
```
where `pViewerView` is a pointer to a `osgViewer::View`.

**Comment by quentinBdr** [comment]
I'll look into that and try to come up with a fix if I find something wrong :) 

Thank you Chris I'll look into that, I might get around the problem somehow checking the mouse orientation !

**Comment by quentinBdr** [comment]
Got a chance to test what chris provided but nothing changed.
Still could not workaround why somehow the Y axis gets inverted before entering ImGui ... But I found a way to overcome that.
By default Qt got its mouse coordinate reference frame from the top left corner of a given widget. I changed that so the reference frame becomes the bottom left corner, thus with Y increasing upwards. So when creating an EventAdaptater I set it like so 
```cpp
if (auto* me = (QMouseEvent*)e)
{
    auto* ea = eventQueue->mouseButtonPress(
        me->x(),
        -me->y() + qwidget->height(),
        qbutton(me->button())
    );
    ea->setMouseYOrientation(osgGA::GUIEventAdapter::Y_INCREASING_UPWARDS);
}
```
With this code, Y stays positive all the way through ImGui, and I did not had to change anything when handling mouse position 
```cpp
io.MousePos = ImVec2(ea.getX(), io.DisplaySize.y - ea.getY());
```
ImGui needs the mouse reference frame from the top left corner (so w/ Y increasing downwards). The bool io.WantCaptureMouse only becomes true if the mouse hovers some ImGui content, so if the mouse coord given is not consistent with ImGui mouse coord system, it stays false.
